poc/four/discovery_client.py

```diff
@@ -0,0 +1,140 @@
+        self.logger = logging.getLogger(__name__)
+        
+    def get_api_metadata(self) -> Dict:
+        """
+        Get API metadata from the Discovery API.
+        
+        Returns:
+            Dictionary containing API metadata
+            
+        Raises:
+            requests.RequestException: If the API request fails
+        """
+        discovery_url = urljoin(self.base_url + '/', 'api')
+        
+        try:
+            response = requests.get(discovery_url)
+            response.raise_for_status()
+            return response.json()
+        except requests.RequestException as e:
+            self.logger.error(f"Failed to get API metadata: {e}")
+            raise
+    
+    def get_oauth_endpoints(self) -> Dict[str, str]:
+        """
+        Get OAuth2 endpoint URLs from the Discovery API.
+        
+        Returns:
+            Dictionary with 'token' and 'authorize' URLs
+            
+        Raises:
+            requests.RequestException: If the API request fails
+            KeyError: If required OAuth endpoints are not found
+        """
+        try:
+            metadata = self.get_api_metadata()
+            
+            # Extract OAuth endpoints from metadata
+            oauth_urls = {}
+            
+            # Look for OAuth URLs in the metadata structure
+            if 'urls' in metadata:
+                for url_info in metadata['urls']:
+                    if 'oauth' in url_info.get('name', '').lower():
+                        oauth_urls['token'] = urljoin(self.base_url + '/', 'oauth/token')
+                        oauth_urls['authorize'] = urljoin(self.base_url + '/', 'oauth/authorize')
+                        break
+            
+            # Fallback to standard OAuth endpoints if not found in metadata
+            if not oauth_urls:
+                oauth_urls = {
+                    'token': urljoin(self.base_url + '/', 'oauth/token'),
+                    'authorize': urljoin(self.base_url + '/', 'oauth/authorize')
+                }
+            
+            self.logger.info(f"Found OAuth endpoints: {oauth_urls}")
+            return oauth_urls
+            
+        except Exception as e:
+            self.logger.error(f"Failed to get OAuth endpoints: {e}")
+            raise
+    
+    def get_data_management_api_base_path(self) -> str:
+        """
+        Get the Data Management API base path.
+        
+        Returns:
+            Base path for Data Management API
+            
+        Raises:
+            requests.RequestException: If the API request fails
+        """
+        try:
+            metadata = self.get_api_metadata()
+            
+            # Look for data management API URL in metadata
+            if 'urls' in metadata:
+                for url_info in metadata['urls']:
+                    if 'data' in url_info.get('name', '').lower():
+                        return url_info.get('url', urljoin(self.base_url + '/', 'data/v3/'))
```

Comment:
Weird, it hard-coded `data/v3` instead of looking for the correct URL. Again, prompting will help.